src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule73wherejavadocrequired/InputMissingJavadocTypeCheckNoViolations.java

```diff
@@ -0,0 +1,72 @@
+package com.google.checkstyle.test.chapter7javadoc.rule73wherejavadocrequired;
+
+/**
+ * This is a Javadoc comment.
+ */
+public class InputMissingJavadocTypeCheckNoViolations { // OK
+
+    /**
+     * This is a Javadoc comment.
+     */
+    public class Inner implements MyInterface{ // OK
+
+    }
+
+    /**
+     * This is a Javadoc comment.
+     */
+    public enum MyEnum { // OK
+
+    }
+
+    /**
+     * This is a Javadoc comment.
+     */
+    public interface MyInterface {  // OK
+
+        /**
+         * This is a Javadoc comment.
+         */
+        class MyInterfaceClass {}  // OK
+    }
+
+    /**
+     * This is a Javadoc comment.
+     */
+    public @interface MyAnnotation { // OK
+
+    }
+
+    /**
+     * This is a Javadoc comment.
+     */
+    protected class InnerProtected { // OK
```

Comment:
It is better to remove javadoc to clearly show that  only public is valudated

Reply:
Protected is too, if it is within a public class; this is why I've added the javadoc here.